keras/src/utils/summary_utils.py

```diff
@@ -96,12 +96,15 @@ def format_shape(shape):
             )
     else:
         try:
-            outputs = layer.compute_output_shape(**layer._build_shapes_dict)
+            if hasattr(layer, "output_shape"):
```

Comment:
@fchollet @james77777778 this breaks [https://github.com/keras-team/keras/blob/v3.6.0/keras/src/layers/attention/multi_head_attention.py#L165](https://github.com/keras-team/keras/blob/v3.6.0/keras/src/layers/attention/multi_head_attention.py#L165) as the multi head attention output_shape's single element array of last dimension size

Reply:
@nglehuy
could you provide a reproducible script for the issue?

Reply:
@james77777778 here you go

```python
import keras


class Model(keras.Model):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.dense = keras.layers.Dense(10)
        self.mha = keras.layers.MultiHeadAttention(10, 10, output_shape=(100,))

    def call(self, inputs):
        x = self.dense(inputs)
        return self.mha(x, x, x)


model = Model()
model(keras.Input(shape=(10, 10)))
model.summary()

```

The errors say

```bash
rich.errors.NotRenderableError: unable to render int; a string or other renderable object is required
```

And when I don't use `output_shape` argument

```python
import keras


class Model(keras.Model):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.dense = keras.layers.Dense(10)
        self.mha = keras.layers.MultiHeadAttention(10, 10)

    def call(self, inputs):
        x = self.dense(inputs)
        return self.mha(x, x, x)


model = Model()
model(keras.Input(shape=(10, 10)))
model.summary()
```

It shows error

```bash
    if len(output_shapes) == 1:
       ^^^^^^^^^^^^^^^^^^
TypeError: object of type 'NoneType' has no len()
```

I use keras==3.6.0

Reply:
I think MultiHeadAttention use the term `output_shape` to indicate the shape of last dimension is kinda incorrect, and it might overlap with the meaning of output_shape of sequential/functional

But changing the argument might cause a break for anyone who using MultiHeadAttention

But if the break is acceptable, I think rename the MultiHeadAttention's `output_shape` is a way to go

Reply:
Got it.
Would you like to submit a PR for this? For now, I think the following snippet can be used to address the issue:

```python
            if hasattr(layer, "output_shape") and not isinstance(
                layer, layers.MultiHeadAttention
            ):
                # `layers.MultiHeadAttention` has an `output_shape` property
                # with a different format from that of `Functional` and
                # `Sequential` models which breaks `summary()`.
                output_shapes = layer.output_shape
```

Reply:
I'm about to work on this and submit the PR but you already did it. Thanks @james77777778 😄